src/app/page.tsx

```diff
@@ -33,15 +34,48 @@ const transactionElement = (transaction: Transaction) => {
 
 export default function Page() {
   const { data, error, isLoading } = useSWR("/api/transactions", fetcher);
+  const [dates, setDates] = useState([new Date(0), new Date()]);
```

Comment:
This default could be problematic if we have a lot of transactions in the DB for this (and started things in 1970 😅 ). But I see we don't really have a way to filter transactions beyond 3 months, so I guess this is fine for now. One to keep an eye on 👀